CONTRIBUTING.md

```diff
@@ -1,15 +1,115 @@
+If you find a bug, want a feature or improvement, have an idea for how to make
+Zanata better, or just want to tell us what you think of something in Zanata,
+please let us know using our [issue system](https://zanata.atlassian.net/)
+
+## <a name="Pull_Requests">Pull Requests</a>
 Pull requests welcome!
 
-General information about our Maven setup is in the wiki:
-https://github.com/zanata/zanata-server/wiki/Developer-Guide
+### <a name="Setup">Setup</a>
+The
+[Developer Setup Guide](https://github.com/zanata/zanata-server/wiki/Developer-Guide)
+shows the dependencies and how to setup Java, Maven, MySQL, JBoss, and IDE.
 
-Our CI server merges changes from `integration/master` to `master` after testing, so: **please branch from `master`, but please target `integration/master` in pull requests.**
+### Branches
+The branches that should be targeted in pull requests
+ - **master**: New features, bug fix, enhancement should target this branch.
+   If you are unsure which branch to make pull request, use this branch.
+ - **release**: Only urgent bug fix and documentation should target this
+ branch.
 
+### Test against Wildfly
 To run all tests against WildFly (takes about 1 hour):
 
-    mvn clean verify -Dappserver=wildfly8 -DstaticAnalysis -Dchromefirefox
+mvn clean verify -Dappserver=wildfly8 -DstaticAnalysis -Dchromefirefox
+
+### <a name="Commit_Message_Format"> Commit Message Format</a>
+We follow the [angular commit message format]
+(https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit),
+that is:
+
+  Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+format that includes a **type**, a **scope** and a **subject**:
+
+```
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+The **header** is mandatory and the **scope** of the header is optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier
+to read on GitHub as well as in various git tools.
+
+#### Type
+Must be one of the following:
+
+- **feat**: A new feature
+- **fix**: A bug fix
+- **docs**: Documentation only changes
+- **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing
+             semi-colons, etc)
+- **refactor**: A code change that neither fixes a bug nor adds a feature
+- **perf**: A code change that improves performance
+- **test**: Adding missing tests
+- **chore**: Changes to the build process or auxiliary tools and libraries such as documentation
+generation
+- **revert**: reverts a previous commit. The subject should be the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+#### Scope
+The scope could be anything specifying place of the commit change. For example `$location`,
+`$browser`, `$compile`, `$rootScope`, `ngHref`, `ngClick`, `ngView`, etc...
```

Comment:
We should use different examples since those are all related to Angular but we aren't using Angular any more (and not at all in the server).

From scrolling through server commits, I found `translation memory`, `user-guide`, `user email`, `password reset`, `statistics`, `applicationConfig`, `glossary`, `contributors` and many fixes that have the bug number as the scope.